runtime/float.go

```diff
@@ -59,6 +59,9 @@ func floatAbs(f *Frame, o *Object) (*Object, *BaseException) {
 }
 
 func floatAdd(f *Frame, v, w *Object) (*Object, *BaseException) {
+	if w.isInstance(ComplexType) {
```

Comment:
I believe the way that CPython works is to let float.\_\_add\_\_ return NotImplemented and then fall back to complex.\_\_radd\_\_. That's how coercion works for other numeric types. So if you implement \_\_radd\_\_ for complex, you can remove these bits of floatAdd and intAdd.